src/discof/replay/fd_replay_tile.c

```diff
@@ -1470,6 +1472,82 @@ handle_new_slice( fd_replay_tile_ctx_t * ctx, fd_stem_context_t * stem ) {
+   - pubkey:         The public key of the vote account to retrieve
+   - stake:          The stake amount associated with this vote account
+   - vote_state_out: Output structure to populate with vote state information
+
+   Failure modes:
+   - Vote account not found in funk database (returns -1)
+   - Vote state cannot be parsed from account data (returns -1)
+   - Vote state has invalid version discriminant (calls FD_LOG_ERR and returns -1)
+   - Vote count exceeds maximum of 31 (calls FD_LOG_ERR, fatal) */
+static int
+fd_replay_out_vote_state_from_funk(
+  fd_funk_t const *            funk,
+  fd_funk_txn_t const *        funk_txn,
+  fd_pubkey_t const *          pubkey,
+  ulong                        stake,
+  fd_replay_out_vote_state_t * vote_state_out ) {
+
+  for(;;) {
+    fd_funk_rec_query_t query;
+    fd_funk_rec_key_t funk_key = fd_funk_acc_key( pubkey );
+    fd_funk_rec_t const * rec = fd_funk_rec_query_try_global( funk, funk_txn, &funk_key, NULL, &query );
+    if( FD_UNLIKELY( !rec ) ) {
+      FD_LOG_WARNING(( "vote account not found. address: %s",
+        FD_BASE58_ENC_32_ALLOCA( pubkey->uc ) ));
+      return -1;
+    }
+
+    fd_voter_state_t const * state = fd_voter_state( funk, rec );
+    if( FD_UNLIKELY( !state ) ) {
+      FD_LOG_WARNING(( "failed to get vote state for vote account %s",
+        FD_BASE58_ENC_32_ALLOCA( pubkey->uc ) ));
+      return -1;
+    }
+
+    /* Speculatively parse the vote state into a replay out vote state */
+    fd_memset( vote_state_out, 0, sizeof(fd_replay_out_vote_state_t) );
+    ulong cnt = fd_voter_state_cnt( state );
+    if( FD_UNLIKELY( fd_funk_rec_query_test( &query ) != FD_FUNK_SUCCESS ) ) continue;
+    if( FD_UNLIKELY( cnt > 31UL ) ) FD_LOG_ERR(( "[%s] funk vote account corruption. cnt %lu > 31", __func__, cnt ));
+
+    vote_state_out->key       = *pubkey;
+    vote_state_out->votes_cnt = cnt;
+    vote_state_out->stake     = stake;
+
+    for( ulong i = 0; i < cnt; i++ ) {
+      if( FD_UNLIKELY( state->discriminant == fd_vote_state_versioned_enum_v0_23_5 ) ) {
+        vote_state_out->votes[i].slot = state->v0_23_5.votes[i].slot;
+        vote_state_out->votes[i].conf = state->v0_23_5.votes[i].conf;
+      } else if( FD_UNLIKELY( state->discriminant == fd_vote_state_versioned_enum_v1_14_11 ) ) {
+        vote_state_out->votes[i].slot = state->v1_14_11.votes[i].slot;
+        vote_state_out->votes[i].conf = state->v1_14_11.votes[i].conf;
+      } else if ( FD_UNLIKELY( state->discriminant == fd_vote_state_versioned_enum_current ) ) {
+        vote_state_out->votes[i].slot = state->votes[i].slot;
+        vote_state_out->votes[i].conf = state->votes[i].conf;
+      } else {
+        FD_LOG_ERR(( "[%s] unknown vote state version. discriminant %u", __func__, state->discriminant ));
+        return -1;
+      }
+    }
+
+    if( FD_LIKELY( fd_funk_rec_query_test( &query ) == FD_FUNK_SUCCESS ) ) {
+      return 0;
+    }
+  }
+}
```

Comment:
is this speculative code still needed? it was this way originally for zero-copy, but really you're now memcpying everything that i think it's much better and less bug-prone to just have a very short speculate -> copy the account bytes -> parse after